helm-charts/basehub/values.schema.yaml

```diff
@@ -492,3 +493,11 @@ properties:
                 type: array
                 items:
                   type: string
+          jupyterhubConfigurator:
+            type: object
+            additionalProperties: false
+            required:
+              - enabled
+            properties:
+              enabled:
+                type: boolean
```

Comment:
I appreciate being able to explicitly disable the configurator whenever it cannot be used. Having it everywhere, especially when it cannot be used is a source of confusion.

I don't think this PR covers it and don't think it should, as it can be its own PR, but I believe we should explicitly disable  it whenever we are enabling profile lists and document this practice

Reply:
I extracted [8c5abc7](https://github.com/2i2c-org/infrastructure/pull/3341/commits/8c5abc72d1073ed8336c944d6aa9d759d3e45d87) into https://github.com/2i2c-org/infrastructure/pull/3348 to enable opt-out of using the configurator as a standalone change for now because I misread this comment.

> I believe we should explicitly disable it whenever we are enabling profile lists and document this practice

I agree that we should disable the configurator (and not explicitly set singleuser.image) if all profileList options override the image.